src/valhalla_path_comparison.cc

```diff
@@ -13,17 +13,144 @@
+  std::cout << "Edge length: " << edge->length() << std::endl;
+  Cost edge_cost = costing->EdgeCost(edge);
+  edge_total += edge_cost;
+  std::cout << "EdgeCost cost: " << edge_cost.cost << " secs: " << edge_cost.secs << "\n";
+  std::cout << "------------------------\n\n";
+}
+
+void walk_edges(const std::string& shape, GraphReader& reader, const std::string& routetype,
+                boost::property_tree::ptree& pt) {
+  // Register edge/node costing methods
+  CostFactory<DynamicCost> factory;
+  factory.Register("auto", CreateAutoCost);
+  factory.Register("auto_shorter", CreateAutoShorterCost);
+  factory.Register("bus", CreateBusCost);
+  factory.Register("bicycle", CreateBicycleCost);
+  factory.Register("hov", CreateHOVCost);
+  factory.Register("motor_scooter", CreateMotorScooterCost);
+  factory.Register("pedestrian", CreatePedestrianCost);
+  factory.Register("transit", CreateTransitCost);
+  factory.Register("truck", CreateTruckCost);
+
+  std::string method_options = "costing_options." + routetype;
+  auto costing_options = pt.get_child(method_options, {});
+  cost_ptr_t cost = factory.Create(routetype, costing_options);
+  TravelMode mode = cost->travel_mode();
+  cost_ptr_t mode_costing[10];
+  mode_costing[static_cast<uint32_t>(mode)] = cost;
+
+  // Decode the shape
+  std::vector<PointLL> shape_pts = decode<std::vector<PointLL> >(shape);
+  if (shape_pts.size() <= 1) {
+    std::cerr << "Not enough shape points to compute the path...exiting" << std::endl;
+  }
+
+  std::vector<Measurement> measurements;
+  for (const auto& ll : shape_pts) {
+    measurements.emplace_back(Measurement{ll, 10, 10});
+  }
+
+  // Add a location for the origin (first shape point) and destination (last
+  // shape point)
+  std::vector<Location> locations;
+  locations.push_back({shape_pts.front()});
+  locations.push_back({shape_pts.back()});
+  const auto projections = Search(locations, reader, cost->GetEdgeFilter(), cost->GetNodeFilter());
+  std::vector<PathLocation> path_location;
+  for (auto loc : locations) {
+    try {
+      path_location.push_back(projections.at(loc));
+    } catch (...) {
+      return;
+    }
+  }
+
+  std::vector<PathInfo> path_infos;
+  std::vector<PathLocation> correlated;
+  bool rtn = RouteMatcher::FormPath(mode_costing, mode, reader, measurements, path_location, path_infos);
+  if (!rtn) {
+    std::cerr << "ERROR: RouteMatcher returned false - did not match complete shape." << std::endl;
+  }
+  GraphId pred_id;
+  GraphId current_id;
+  uint64_t current_osmid = 0;
+  Cost edge_total;
+  Cost trans_total;
+  for (const auto& path_info : path_infos) {
+    //std::cout << "lat: " << result.lnglat.lat() << " lon: " << result.lnglat.lng() << std::endl;
+    if (path_info.edgeid == current_id || path_info.edgeid == kInvalidGraphId) {
+      continue;
+    }
+
+    current_id = path_info.edgeid;
+    print_edge(reader, cost, current_id, pred_id, edge_total, trans_total, current_osmid);
```

Comment:
if in the end, you just need edge ids, why not just instantiate an `actor` can called trace_attributes, either with encoded shape or whatever and use the return to get the edges? it would be a ton less code and it would ensure using the same code we use for the actual services instead of recreating it. maybe youre getting something here other than just edge ids though? im not sure.